main/SQLHelper.cpp

```diff
@@ -5176,6 +5194,100 @@ void CSQLHelper::UpdateUVLog()
 	}
 }
 
+bool CSQLHelper::UpdateCalendarMeter(const int HardwareID, const char* DeviceID, const unsigned char unit, const unsigned char devType, const unsigned char subType, bool dayCalendar, float counter, float usage, const char* date)
+{
+	std::vector<std::vector<std::string> > result;
+	char szTmp[200];
+
+	result = safe_query("SELECT ID, Name, SwitchType FROM DeviceStatus WHERE (HardwareID=%d AND DeviceID='%q' AND Unit=%d AND Type=%d AND SubType=%d)", HardwareID, DeviceID, unit, devType, subType);
+
+	if (result.empty()) {
+		return false;
+	}
+
+	uint64_t DeviceRowID;
+	
+	if (counter < 0.0) {
+		counter = 0.0;
+	}
+	if (usage < 0.0) {
+		usage = 0.0;
+	}
+
+	std::vector<std::string> sd = result[0];
+	std::stringstream s_strid;
+	s_strid << sd[0];
+	s_strid >> DeviceRowID;
+	std::string devname = sd[1];
+	_eSwitchType switchtype=(_eSwitchType) atoi(sd[2].c_str());
+
+	if (dayCalendar)
```

Comment:
this can not be the right boolean for the Meter table, the meter table is for short-log, and tables ending with _Calendar are for day values

Reply:
The name of the boolean is misleading, I will change it